radicle/src/ssh.rs

```diff
@@ -0,0 +1,285 @@
+pub mod agent;
+
+use std::io;
+use std::mem;
+use std::ops::DerefMut;
+
+use byteorder::{BigEndian, WriteBytesExt};
+use thiserror::Error;
+use zeroize::Zeroizing;
+
+use radicle_ssh::encoding;
+use radicle_ssh::encoding::Encoding as _;
+
+use crate::crypto;
+use crate::crypto::PublicKey;
+
+pub mod fmt {
+    use crate::crypto::PublicKey;
+
+    /// Get the SSH long key from a public key.
+    /// This is the output of `ssh-add -L`.
+    pub fn key(key: &PublicKey) -> String {
+        use byteorder::{BigEndian, WriteBytesExt};
+
+        let mut buf = Vec::new();
+        let key = key.as_ref();
+        let len = key.len();
+
+        buf.write_u32::<BigEndian>(len as u32)
+            .expect("Writing to vectors doesn't fail");
+        buf.extend_from_slice(key);
+
+        // Despite research, I have no idea what this string is, but it seems
+        // to be the same for all Ed25519 keys.
+        let mut encoded = String::from("ssh-ed25519 AAAAC3NzaC1lZDI1NTE5");
```

Comment:
I looked this up myself. Here's what I found[[0]][[1]].

It looks like it's due to this[[2]], so the value is calculated as:
```
$ echo -ne "\0\0\0\x0bssh-ed25519" | base64
AAAAC3NzaC1lZDI1NTE5
```

[0]: https://forums.unraid.net/bug-reports/stable-releases/6103-ui-field-for-root-ssh-authorized-keys-does-not-support-ed25519-and-other-valid-key-types-r2023/
[1]: https://github.com/nemchik/ssh-key-regex
[2]: https://github.com/nemchik/ssh-key-regex#description

Reply:
ahah! We can implement it that way perhaps, nice find.